### PR TITLE
Pin, unpin, update now take &Cid

### PR DESCRIPTION
**Pull request by troublescooter**

### CODE REVIEW

**Review by dvc94ch — COMMENTED** [review]
Can you make it  T: Into<Cow<'a, Cid>>

**Comment by troublescooter** [comment]
Tests contingent on [conversion traits](https://github.com/ipfs-rust/tiny-cid/pull/2). You'll have to bump versions.

File: src/store.rs [hunk 1]
```diff
@@ -76,19 +77,19 @@ impl<S: StoreParams> Transaction<S> {
     }
 
     /// Increases the pin count of a block.
-    pub fn pin(&mut self, cid: Cid) {
+    pub fn pin(&mut self, cid: Cow<'a, Cid>) {
         self.ops.push_back(Op::Pin(cid));
```
Comment (dvc94ch) [inline src/store.rs]: Why not use Into Cow here?

**Review by dvc94ch — COMMENTED** [review]
Tests are failing

**Pull request merged**